pyresample/_formatting_html.py

```diff
@@ -0,0 +1,361 @@
+
+    if icon is None:
+        icon = _icon("icon-database")
+
+    return ("<div class='pyresample-area-section-item'>"
+            f"<input id='{data_id}' class='pyresample-area-section-in' "
+            f"type='checkbox' {enabled} {collapsed}>"
+            f"<label for='{data_id}' {tip}>{icon} {name}</label>"
+            f"<div class='pyresample-area-section-preview'>{inline_details}</div>"
+            f"<div class='pyresample-area-section-details'>{details}</div>"
+            "</div>"
+            )
+
+
+def map_section(area):
+    """Create html for map section.
+
+    Args:
+        area(Union[:class:`~pyresample.geometry.AreaDefinition`, :class:`~pyresample.geometry.SwathDefinition`]):
+            Area definition or Swath definition.
+
+    Returns:
+        str: String of html.
+
+    """
+    map_icon = _icon("icon-globe")
+
+    if cart:
+        coll = collapsible_section("Map", details=plot_area_def(area, fmt="svg"), collapsed=True, icon=map_icon)
+    else:
+        coll = collapsible_section("Map",
+                                   details="Note: If cartopy is installed a display of the area can be seen here",
+                                   collapsed=True, icon=map_icon)
+
+    return f"{coll}"
+
+
+def proj_area_attrs_section(area):
+    """Create html for attribute section based on an area Area.
+
+    Args:
+        area (:class:`~pyresample.geometry.AreaDefinition`):
+            Area definition.
+
+    Returns:
+        str: String of html.
+
+    """
+    resolution_str = "/".join([str(round(x, 1)) for x in area.resolution])
+    proj_dict = area.proj_dict
+    proj_str = "{{{}}}".format(", ".join(["'%s': '%s'" % (str(k), str(proj_dict[k])) for k in
+                                          sorted(proj_dict.keys())]))
+    area_units = proj_dict.get("units", "")
+
+    attrs_icon = _icon("icon-file-text2")
+
+    area_attrs = ("<dl>"
+                  f"<dt>Area name</dt><dd>{area.area_id}</dd>"
+                  f"<dt>Description</dt><dd>{area.description}</dd>"
+                  f"<dt>Projection</dt><dd>{proj_str}</dd>"
+                  f"<dt>Width/Height</dt><dd>{area.width}/{area.height} Pixel</dd>"
+                  f"<dt>Resolution x/y (SSP)</dt><dd>{resolution_str} {area_units}</dd>"
+                  f"<dt>Extent (ll_x, ll_y, ur_x, ur_y)</dt>"
+                  f"<dd>{tuple(round(x, 4) for x in area.area_extent)}</dd>"
+                  "</dl>"
+                  )
+
+    coll = collapsible_section("Properties", details=area_attrs, icon=attrs_icon)
+
+    return f"{coll}"
+
+
+def swath_area_attrs_section(area):
+    """Create html for attribute section based on SwathDefinition.
+
+    Args:
+        area (:class:`~pyresample.geometry.SwathDefinition`): Swath definition.
+
+    Returns:
+        str: String of html.
+
+    Todo:
+        - Improve resolution estimation from lat/lon arrays. Maybe use CoordinateDefinition.geocentric_resolution?
```

Comment:
Better resolution and boundary information is something I'd say is involved with ghiggi's PRs (not tagging just to avoid spam from the rest of the conversation) related to boundaries. I think swath definitions having a more accurate resolution will have to come later. Something I hope comes to Satpy and pyresample in the future where things like VIIRS and MODIS readers add g-ring/polygon information to the SwathDefinitions and along with that more specific estimates of limb and nadir resolutions. It is added in some readers as metadata to the lons/lats DataArrays but definitely isn't required.

Reply:
Yes I think we already talked about that somewhere. Just put it here as a reminder that this could use some improvement because it is not very obvious from the code.